### PR TITLE
WIP Names and Types variable

### PR DESCRIPTION
**Pull request by PavelBoPavel**

### CODE REVIEW

**Review by lamaconda — DISMISSED** [review]
lgfm

File: homework/PavelBo/dataTypes/encoder/src/main/java/com/github/pavelbopavel/EncoderDemo.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+public class EncoderDemo {
+    static Encoder encoder = new Encoder();
+
+    public static void main(String[] args) {
+        short code = 65;
+        encoder.decoder('X');
+        encoder.encoder(code);
+    }
+}
```
Comment (max-java) [inline homework/PavelBo/dataTypes/encoder/src/main/java/com/github/pavelbopavel/EncoderDemo.java:11]: this should be moved to test

File: homework/PavelBo/dataTypes/nameVariables/src/main/java/com/github/PavelBoPavel/bean/TypesVariable.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    private int speedCar;
+    private int integerUserEntered;
+    private int largestInteger;
+    private int smallestRealNumber;
+    private int leftBorderOfRangeOfNumbers;
+    private int rightBorderOfRangeOfNumbers;
+    private int integerUserGuessed;
+    private double cashBalance;
+    private int numberOfWheelsOnBike;
+    private boolean hungryMan;
```
Comment (max-java) [inline homework/PavelBo/dataTypes/nameVariables/src/main/java/com/github/PavelBoPavel/bean/TypesVariable.java:20]: isHungry will be more more obvious

**Review by max-java — CHANGES_REQUESTED** [review]
Technically looks good, but I insist that you should use quick start template and sonarCube analyse. 
Please, make appropriate changes and add tests.

File: homework/PavelBo/dataTypes/nameVariables/src/main/java/com/github/PavelBoPavel/bean/NamesVariable.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+public class NamesVariables {
+
+    private int numberOfShipsInPort;
+    private int counterInteger;
+    private int firstNumber;
+    private String answerTheQuestion;
+    private double randomNumber;
+    private double randomNumberGenerator;
+    private String carColor;
+    private int speedCar;
```
Comment (KaydunovDenis) [inline homework/PavelBo/dataTypes/nameVariables/src/main/java/com/github/PavelBoPavel/bean/NamesVariable.java:12]: carSpeed

**Review by KaydunovDenis — APPROVED** [review]

**Review by trump-ny — DISMISSED** [review]

**Review by Puzzletania — DISMISSED** [review]

**Review by lamaconda — APPROVED** [review]
lgfm

**Review by VasiliZas — APPROVED** [review]

File: homework/PavelBo/collections/library/src/main/java/com/tutrit/java/quickstart/Book.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    public String getAuthorBook() {
+        return authorBook;
+    }
+
+    public String getNameBook() {
+        return nameBook;
+    }
+}
```
Comment (lamaconda) [inline homework/PavelBo/collections/library/src/main/java/com/tutrit/java/quickstart/Book.java:43]: Я бы сделал первым конструктор, потом геттеры сеттеры, потом equals and hash code, и в конце ту стринг

File: homework/PavelBo/collections/library/src/test/java/com/tutrit/java/quickstart/LibraryTest.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        bookChehov = new Book("Chehov", "chehiy");
+        bookTolstoy = new Book("Tolstoy", "tolstolobik");
+        bookTolstoy2 = new Book("Tolstoy", "karas");
+        library.addBook(bookPuskin);
+        library.addBook(bookPuskin2);
+        library.addBook(bookPuskin3);
+        library.addBook(bookChehov);
+        library.addBook(bookTolstoy);
+        library.addBook(bookTolstoy2);
+    }
```
Comment (lamaconda) [inline homework/PavelBo/collections/library/src/test/java/com/tutrit/java/quickstart/LibraryTest.java:36]: необходимо ли всё это делать перед каждым тестом?

**Review by lamaconda — APPROVED** [review]
lgfm

**Review by KaydunovDenis — DISMISSED** [review]

File: homework/PavelBo/collections/library/src/test/java/com/tutrit/java/quickstart/LibraryTest.java (hunk 5 above)
Comment (PavelBoPavel) [reply homework/PavelBo/collections/library/src/test/java/com/tutrit/java/quickstart/LibraryTest.java:36]: да. Это используется каждый раз

File: homework/PavelBo/dataTypes/encoder/src/main/java/com/github/pavelbopavel/EncoderDemo.java (hunk 1 above)
Comment (PavelBoPavel) [reply homework/PavelBo/dataTypes/encoder/src/main/java/com/github/pavelbopavel/EncoderDemo.java]: так и сделал, чуть ниже в комите "@PavelBoPavel
refactoring encoder. Add test"

File: homework/PavelBo/collections/stringTokenizer/src/main/java/com/tutrit/java/quickstart/StringAnalyzer.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+                        int value = punctuationMark.getValue() + 1;
+                        punctuationMark.setValue(value);
+                    }
+                });
+            } else {
+                punctuationMarksMap.put(punctuationMarks, 1);
+            }
+        }
+    }
+}
```
Comment (max-java) [inline homework/PavelBo/collections/stringTokenizer/src/main/java/com/tutrit/java/quickstart/StringAnalyzer.java:66]: this looks complicated. Could you please split this into separate method that could be tested independently?

File: homework/PavelBo/collections/stringTokenizer/src/test/java/com/tutrit/java/quickstart/StringAnalyzerTest.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+import junit.framework.Assert;
+import junit.framework.TestCase;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Set;
+
+public class StringAnalyzerTest extends TestCase {
```
Comment (max-java) [inline homework/PavelBo/collections/stringTokenizer/src/test/java/com/tutrit/java/quickstart/StringAnalyzerTest.java:11]: use jUnit4 or 5

File: homework/PavelBo/collections/stringTokenizer/src/test/java/com/tutrit/java/quickstart/StringAnalyzerTest.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Before
+    public void setUp() {
+        stringAnalyzer = new StringAnalyzer();
+    }
+
+    @Test
+    public void testSplitToSentences() {
+        List<String> sentences = stringAnalyzer.splitToSentences(Text.text);
+        Assert.assertEquals(17, sentences.size());
```
Comment (max-java) [inline homework/PavelBo/collections/stringTokenizer/src/test/java/com/tutrit/java/quickstart/StringAnalyzerTest.java:23]: Bp is to use static imports for Assert methods

File: homework/PavelBo/collections/stringTokenizer/src/test/java/com/tutrit/java/quickstart/StringAnalyzerTest.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @Test
+    public void testSplitToWords (){
+        List<String> words = stringAnalyzer.splitToWords(Text.text);
```
Comment (max-java) [inline homework/PavelBo/collections/stringTokenizer/src/test/java/com/tutrit/java/quickstart/StringAnalyzerTest.java:28]: It's better not to use production data in unit tests, but simple data that obviously shows that method works. Replace Text.text with something simple

File: homework/PavelBo/collections/word vocabulary/src/main/java/com/github/pavelbopavel/UniqueWordVocabulary.java [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+public class UniqueWordVocabulary {
+    Set<String> words = new HashSet<>();
+    Logger log = LoggerFactory.getLogger(UniqueWordVocabulary.class);
+
+    public void addWord(String word) {
+        if(word.equals("")){}
+        else {
+            words.add(word);
+
+        }
```
Comment (max-java) [inline homework/PavelBo/collections/word vocabulary/src/main/java/com/github/pavelbopavel/UniqueWordVocabulary.java:22]:
```
if(!word.equals("")) {
  words.add(word);
}
```

File: homework/PavelBo/collections/word vocabulary/src/main/java/com/github/pavelbopavel/UniqueWordVocabulary.java [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+        words.forEach(w -> log.info(w));
+    }
+
+    UniqueWordVocabulary uniqueWordVocabulary;
+    @Before
+    public void setUp() {
+        uniqueWordVocabulary = new UniqueWordVocabulary();
+        uniqueWordVocabulary.words.add("hello");
+        uniqueWordVocabulary.words.add("dog");
+        uniqueWordVocabulary.words.add("cat");
```
Comment (max-java) [inline homework/PavelBo/collections/word vocabulary/src/main/java/com/github/pavelbopavel/UniqueWordVocabulary.java:39]: try Set.of()

File: homework/PavelBo/collections/word vocabulary/src/main/java/com/github/pavelbopavel/UniqueWordVocabulary.java [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+        assertEquals(3,uniqueWordVocabulary.getWordsCount());
+
+    }
+
+    @Test
+    public void printVocabulary_Test() {
+        uniqueWordVocabulary.printVocabulary();
+    }
+}
+
```
Comment (max-java) [inline homework/PavelBo/collections/word vocabulary/src/main/java/com/github/pavelbopavel/UniqueWordVocabulary.java:63]: That should be moved in test!

File: homework/PavelBo/collections/wordCounter/src/test/java/com/tutrit/java/quickstart/UniqueWordCounterTest.java [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+import junit.framework.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static junit.framework.Assert.*;
+import static org.junit.Assert.*;
```
Comment (max-java) [inline homework/PavelBo/collections/wordCounter/src/test/java/com/tutrit/java/quickstart/UniqueWordCounterTest.java:11]: you'd better avoid of use * in imports

File: homework/PavelBo/collections/wordCounter/src/test/java/com/tutrit/java/quickstart/UniqueWordCounterTest.java [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+    @Test
+    public void addWord_test(){
+        uniqueWordCounter.addWord("hello");
+        uniqueWordCounter.addWord("dog");
+        uniqueWordCounter.addWord("cat");
+        uniqueWordCounter.addWord("hello");
+        uniqueWordCounter.addWord("hello");
+        uniqueWordCounter.addWord("dog");
+
+    }
```
Comment (max-java) [inline homework/PavelBo/collections/wordCounter/src/test/java/com/tutrit/java/quickstart/UniqueWordCounterTest.java:38]: you should add at least 1 assertion

File: homework/PavelBo/dataTypes/encoder/src/test/java/com/github/pavelbopavel/EncoderTest.java [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+
+    @Test
+    public void decoder() {
+        encoder.decoder('X');
+    }
+
+    @Test
+    public void encoder() {
+        short code = 65;
+        encoder.encoder(code);
```
Comment (max-java) [inline homework/PavelBo/dataTypes/encoder/src/test/java/com/github/pavelbopavel/EncoderTest.java:25]: you should have at lease 1 assertion

for that, you should return a value )

File: homework/PavelBo/dataTypes/wrappers/src/test/java/com/github/pavelbopavel/WrapperDemoTest.java [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class WrapperDemoTest {
+    Logger log;
+
+    @Before
+    public void setUp(){
+        log = LoggerFactory.getLogger(WrapperDemoTest.class);
```
Comment (max-java) [inline homework/PavelBo/dataTypes/wrappers/src/test/java/com/github/pavelbopavel/WrapperDemoTest.java:15]: I don't think we want to logger will be created for every test, do we?

File: homework/PavelBo/dataTypes/wrappers/src/test/java/com/github/pavelbopavel/WrapperDemoTest.java [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+    }
+    @Test
+    public void immutableDemo(){
+        Integer i1 = 10;
+        Integer i2 = i1;
+        Assert.assertEquals(i1.hashCode(), i2.hashCode());
```
Comment (max-java) [inline homework/PavelBo/dataTypes/wrappers/src/test/java/com/github/pavelbopavel/WrapperDemoTest.java:21]: that is not how immutability tests. 
When you test if object hasn't been mutate, you should prove that it's original value did not changed. Revise your test

**Review by max-java — CHANGES_REQUESTED** [review]
ptal

Please, one merge request for one task

**Review by VasiliZas — APPROVED** [review]

**Pull request closed without merging**